Issue 173/delete document styling updates

andycwilliams opened the pull request:
Applying styling to the Delete Document component. Nearly identical to the styling done on #131. 

Note how the document selection remains unstyled, just as with Search Document. As it is an imported function that affects multiple components, it will be changed once those have had their initial styling. This is to limit the possibility of errors and breaking functionality.

![2023-05-23 (1)](https://github.com/codeforpdx/PASS/assets/83156697/994d7ff9-6407-411d-ac4c-2afb29846384)

andycwilliams committed:
Implementing styling from FetchDocumentForm

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

leekahung committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

leekahung commented on src/components/Form/DeleteDocumentForm.jsx:
```diff
@@ -50,30 +55,35 @@ const DeleteDocumentForm = () => {
-        </div>
-      </form>
+      <Box sx={{ minWidth: 120 }}>
+        <form onSubmit={handleDeleteDocument} autoComplete="off">
```
Now that I think about it, I don't think there are text field for this form or for FetchDocumentForm, they're already pre-populated.

Think we can safely remove the autoComplete attributes without affecting anything.

andycwilliams replied:
Removed.

leekahung reviewed (changes requested):
Nothing too big, just one more thing. See feedback.

timbot1789 reviewed (commented):
Why are the buttons stretched to the max width?

andycwilliams committed:
Removing autocomplete

andycwilliams committed:
Merge branch 'issue-173/delete-document-styling-updates' of https://github.com/codeforpdx/PASS into issue-173/delete-document-styling-updates

leekahung commented on src/components/Form/FetchDocumentForm.jsx:
```diff
@@ -67,11 +67,11 @@ const FetchDocumentForm = () => {
             <i>Select Document Type</i>
           </InputLabel>
-          <FormControl fullWidth variant="standard">
+          <FormControl  variant="standard">
```
Prettier caught the extra space here. do a `npm run prettier:run` to fix it.

andycwilliams replied:
When I run that I get this error:
```
[error] No files matching the pattern were found: "'src/**/*.js'".
[error] No files matching the pattern were found: "'src/**/*.jsx'".
[error] No files matching the pattern were found: "'test/**/*.js'".
[error] No files matching the pattern were found: "'test/**/*.jsx'".
```

leekahung replied:
Not sure what happened, but this is what I get when I ran your latest changes. Seems to catch them just fine.

<img width="1440" alt="Screen Shot 2023-05-24 at 23 09 27" src="https://github.com/codeforpdx/PASS/assets/14917816/baf1c0f0-3083-4a1c-8843-2ead3f9cf4de">

leekahung replied:
Still thinking about what might have gone wrong on your end with Prettier, but the space problem is gone, so I'm fine for now.

andycwilliams replied:
Doesn't seem to be a real issue, but I wanted to provide more information on this in case it's useful to anyone.

Running `npm run prettier:run` yields the same error.

However, when I run `npm run prettier:check`, I get this: 

```
Checking formatting...
[error] No files matching the pattern were found: "'src/**/*.js'".
[error] No files matching the pattern were found: "'src/**/*.jsx'".
[error] No files matching the pattern were found: "'test/**/*.js'".
[error] No files matching the pattern were found: "'test/**/*.jsx'".
All matched files use Prettier code style!
```

The only explanation in my mind is that since I have Prettier installed on my editor anyway, the script checks for that and does not execute properly.

leekahung replied:
Also, refrain from closing conversations other people started. In this case, other developers might have the same issue after all. This way, they could check up on the progress.

leekahung commented on src/components/Form/FetchDocumentForm.jsx:
Like what we did with the form containers, I think we can center the form elements if we're going to shrink the buttons to an appropriate size.

![Screen Shot 2023-05-24 at 0 25 57](https://github.com/codeforpdx/PASS/assets/14917816/b62b5342-0dbb-4690-a8a5-2f91c8142ece)

We can do the same thing for the "Delete Document". Think it'll turn out nicely.

leekahung reviewed (dismissed):
See feedback for request changes.

andycwilliams committed:
Centering form elements

andycwilliams commented:
Tweaked the styling so that the form elements are smaller and centered. There is still a large amount of empty space but we're still waiting on guidance from the UX team on how to format the desktop version.

Also, I'm not sure how we will end up implementing the status updates at the bottom of the forms. Will they remain there, appear in a pop-up, something else?

![2023-05-24 (3)](https://github.com/codeforpdx/PASS/assets/83156697/8b2c2377-c11c-470b-8e18-cffae7c3fe92)

xscottxbrownx commented:
> Tweaked the styling so that the form elements are smaller and centered. There is still a large amount of empty space but we're still waiting on guidance from the UX team on how to format the desktop version.
> 
> Also, I'm not sure how we will end up implementing the status updates at the bottom of the forms. Will they remain there, appear in a pop-up, something else?
> 
> ![2023-05-24 (3)](https://user-images.githubusercontent.com/83156697/240795363-8b2c2377-c11c-470b-8e18-cffae7c3fe92.png)

I was thinking about these the other day...

my approach without any input from others would've been to have a spinning loader show while the function was running, once the function is complete it will show the result.  But that (once function is complete it will show result) would depend a little on what function/form it is and how they are displayed on the page.

Something that needs a discussion for sure.

xscottxbrownx commented on src/components/Form/DeleteDocumentForm.jsx:
```diff
@@ -50,30 +55,29 @@ const DeleteDocumentForm = () => {
+            <i>Select Document Type to Delete:</i>
+          </InputLabel>
+          <FormControl variant="standard" fullWidth>
+            <DocumentSelection htmlId="search-doctype" />
```
**line 71** - `htmlID` should be "delete-doctype" to match the `id` of the label to link the two together.

leekahung replied:
Good catch!

xscottxbrownx reviewed (changes requested):
just the tiny change for the label and it's matching input, then looks good to me.

andycwilliams committed:
Fixing htmlId

andycwilliams commented:
> I was thinking about these the other day...
> 
> my approach without any input from others would've been to have a spinning loader show while the function was running, once the function is complete it will show the result. But that (once function is complete it will show result) would depend a little on what function/form it is and how they are displayed on the page.
> 
> Something that needs a discussion for sure.

I've tested using a loading animation for the UsersList since it takes a moment longer to load than everything else, and I think it looks good.

We need more feedback before changing it, but a quick pop-up animation or alert makes sense to me. At least for some of these.

leekahung committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

leekahung commented:
> ![2023-05-24 (3)](https://user-images.githubusercontent.com/83156697/240795363-8b2c2377-c11c-470b-8e18-cffae7c3fe92.png)

This is starting to look more modal-like, which is good if we're going to transition to it. One thing we could do too, now that we're talking about the status notification is that we can center this too so that they're all lined up. The animation/message could replace what we have now for the purple status text eventually.

leekahung reviewed (commented):
Minor stylistic comments. See feedback.

andycwilliams committed:
Beginning restyling status with MUI

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

andycwilliams committed:
Merge branch 'issue-173/delete-document-styling-updates' of https://github.com/codeforpdx/PASS into issue-173/delete-document-styling-updates

andycwilliams committed:
Formatting progress bars., incl. Users List

andycwilliams committed:
Centering more elements

andycwilliams commented:
Alright, so I've centered the elements more and refactored a bit of styling from the CSS file into the MUI theme. Plus I did a bit of formatting with the forms, mostly keeping in mind the modal approach. It ought to be easier to go from modal to desktop than vice versa.

![2023-05-26 (3)](https://github.com/codeforpdx/PASS/assets/83156697/bfcbf32a-25da-463e-9d6a-00d2d34a9263)

![2023-05-26 (2)](https://github.com/codeforpdx/PASS/assets/83156697/fd2cfcb6-9d03-41cd-b815-aacaa8f3c797)

Another change is adding loading animations. That wasn't a necessity for this PR, but I think this is a decent start. I added one to the Users List as well. By no means do we have to keep that. I just think it looks nice and wanted to share it.

![2023-05-26 (1)](https://github.com/codeforpdx/PASS/assets/83156697/7aa0d31e-1d86-4a2e-9b50-d1e67b994ec4)

I also tried to shorten the status messages a bit because they really stretch out the forms otherwise.

In addition, a lot of the smaller changes such as spacing, minimum width, text styling, etc. can be easily tweaked. Primarily I wanted to get more files converted to MUI, so a few more have been changed than only Delete Document.

There's certainly refactoring to do and redundant code to trim, but with this commit I'm looking for feedback on whether this is the direction we want to go.

leekahung commented on src/components/Notification/StatusMessage.jsx:
I certainly like the loading animations, we probably would need to tweak the timing for StatusMessage or when we should use the loading animation for StatusMessage. For something like uploading documents or searching documents, this works perfectly. The examples you showed in your last comment are good examples of when to use them as well. Would like to keep them if we can, however, there are a few edge cases.

For that reason, I would like to hold on just this one thing.

For example, for the selection of users, the "select user" interaction with the button should be instantaneous since it's all within PASS via React Context and not between PASS and Solid (which takes a significantly longer amount of time). So a loading animation would be unnecessary. But because we're using StatusMessage to indicate that, the loading animation is unfortunately triggered. Think we could refactor this to use an alternative message scheme (but that's for another PR).

Another case where we might not need the animation is when an operation errors out.

![Screen Shot 2023-05-26 at 18 11 48](https://github.com/codeforpdx/PASS/assets/14917816/d493d99c-65cf-4ac6-a04f-c03740c26f0f)

While it makes sense to disable the button to prevent redundant actions being performed while the function is running, it shouldn't still still be a "loading" screen if we know the operation failed. We can perhaps show an alternative icon/animation.

I certainly want these animations, but I think should do this in another PR where it can be better flushed out.

leekahung replied:
Planning to approve without the loading animation with StatusMessage/StatusNotification at this time.

leekahung reviewed (commented):
See feedback comment.

leekahung commented on src/components/Notification/StatusMessage.jsx:
```diff
@@ -18,19 +21,28 @@ import React from 'react';
       {notification}
-    </dd>
+      <br />
+      <CircularProgress />
```
I think we can just comment this out in the component and add a comment on top as a `TODO: need to discuss when to trigger animation`. That means we would need to comment out the import for `<CircularProgress />` as well. At least for the time being.

That way we can keep what we have in this PR and not load the animation until we figure out what to do with them.

andycwilliams committed:
Troubleshooting validateDomNesting error

leekahung committed:
Reverting to original changes to avoid merge conflict with package-lock.json

andycwilliams committed:
Merge branch 'issue-173/delete-document-styling-updates' of https://github.com/codeforpdx/PASS into issue-173/delete-document-styling-updates

leekahung commented on src/components/Notification/StatusNotification.jsx:
```diff
@@ -19,14 +21,22 @@ import StatusMessage from './StatusMessage';
 const StatusNotification = ({ notification, statusType, defaultMessage, locationUrl = '' }) => (
-  <dl className="display">
-    <dt>{statusType}:</dt>
+  <Typography
```
Line 24 `<Typogrphy>` component has a `<div>` inside it (Line 37). This will trigger an error in the console log. Switch this to perhaps a `<Box>`.

leekahung replied:
Line 24 `<Typogrphy>` component has a `<div>` inside it (Line 37). This will trigger an error in the console log. Switch this to perhaps a `<Box>`.

leekahung commented on src/components/Notification/StatusNotification.jsx:
```diff
@@ -19,14 +21,22 @@ import StatusMessage from './StatusMessage';
+      alignItems: 'center'
+    }}
+  >
+    <>{statusType}:</>
```
Considering it was suppose to be a `<dt>` element, you could wrap this with `<Typography>`

leekahung commented on src/components/Users/UserSection.jsx:
```diff
@@ -23,12 +29,26 @@ const UserSection = ({ loadingUsers, loadingActive }) => {
+            }}
+          >
+            <Paper elevation={2} sx={{ display: 'inline-block', mx: '2px', padding: '20px' }}>
+              <Typography sx={{ marginBottom: '20px', textAlign: 'center' }}>
```
The `<Typography>` component in Line 42 will trigger an error since there's a `<div>` inside it at Line 46. Could replace this with a `<Box>`.

leekahung commented on src/style.css:
```diff
@@ -34,16 +34,12 @@ header {
 }
 
-.display {
+/* .display {
```
I think we can remove these unused stylings if we're replacing them with MUI components used in this PR.

leekahung reviewed (changes requested):
See feedback for updates.

andycwilliams committed:
Resolving MUI Typography errors

leekahung committed:
Reverting package-lock.json back

leekahung commented on src/components/Form/DeleteDocumentForm.jsx:
```diff
@@ -50,30 +55,29 @@ const DeleteDocumentForm = () => {
+          <InputLabel id="delete-doctype">
+            <i>Select Document Type to Delete:</i>
+          </InputLabel>
+          <FormControl variant="standard" fullWidth>
```
This is good. Since we're starting to add `<FormControl>` around DocumentSelection, we should be able to edit DocumentSelection soon with MUI `<Select>`/`<MenuItem>` instead of HTML `<select>`/`<option>`. Should be able to do it once we get the other forms with DocumentSelection moved over to MUI. We'll leave that to a different PR.

leekahung commented on src/components/Form/DeleteDocumentForm.jsx:
```diff
@@ -50,30 +55,29 @@ const DeleteDocumentForm = () => {
+      <Box display="flex" justifyContent="center" sx={{ minWidth: 120 }}>
+        <form onSubmit={handleDeleteDocument}>
+          <InputLabel id="delete-doctype">
+            <i>Select Document Type to Delete:</i>
```
Think we should use `<em>` instead for emphasizing text like how we're using `<strong>` instead of `<b>`.

leekahung commented on src/components/Notification/StatusNotification.jsx:
```diff
@@ -19,14 +22,33 @@ import StatusMessage from './StatusMessage';
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center'
+      }}
```
These should be in `<Box>` instead of `<Typography>`.

leekahung commented on src/components/Notification/StatusNotification.jsx:
```diff
@@ -19,14 +22,33 @@ import StatusMessage from './StatusMessage';
+          flexDirection: 'column',
+          justifyContent: 'center',
+          alignItems: 'center'
+        }}
```
We can remove Lines 41 to 46 since the styling of the `Box` contents would be handled by `<Box>` in Line 25 once you move the sx from Line 27 to 33 into it.

![Screen Shot 2023-05-27 at 9 48 11](https://github.com/codeforpdx/PASS/assets/14917816/201a85bb-326a-478d-b3ff-4083ee8ac63f)

![Screen Shot 2023-05-27 at 9 50 28](https://github.com/codeforpdx/PASS/assets/14917816/87c70318-ebf7-4910-8e87-b1d0703f475b)

leekahung commented on src/theme.js:
```diff
@@ -2,7 +2,15 @@ import { createTheme } from '@mui/material/styles';
 const theme = createTheme({
   typography: {
-    fontFamily: 'Roboto, sans-serif'
+    // button: {
```
Can remove these since they're unused

leekahung commented on src/components/Notification/StatusMessage.jsx:
```diff
@@ -18,19 +24,35 @@ import React from 'react';
   return (
-    <dd className="labelStatus" role="alert">
+    <Typography
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center'
+      }}
```
Lines 45 to 50 can be removed. Styling for centering the text is already handled by StatusNotification via `<Box>`.

leekahung commented on src/components/Notification/StatusMessage.jsx:
```diff
@@ -18,19 +24,35 @@ import React from 'react';
-          {locationUrl}
+      <Typography
+        variant="dd"
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          alignItems: 'center'
+        }}
```
Lines 29 to 33 can be removed. Styling for centering the text is already handled by StatusNotification via `<Box>`.

leekahung reviewed (changes requested):
See requested feedback of the latest changes.

Additionally run a git pull for this update before doing your changes again. It looks like there's a rogue package-lock.json file in there, which shouldn't happen with this PR unless package-lock.json was delete and regenerated with npm install. So I've copied a good package-lock.json file from your branch a few commits back and pushed it back into the branch.

andycwilliams committed:
Consolidating styling into MUI Box component

andycwilliams committed:
Merge branch 'Development' into issue-173/delete-document-styling-updates

leekahung commented on src/components/Users/UserSection.jsx:
```diff
@@ -21,14 +27,28 @@ const UserSection = ({ loadingUsers, loadingActive }) => {
-          <div style={{ marginBottom: '20px', textAlign: 'center' }}>Loading users list...</div>
-        </section>
-      ) : (
         <UsersList loadingActive={loadingActive} />
```
Minor thing, I think I can update this since it's close to being finished. But the ordering of this ternary should be reversed.

If loadingUsers is true, we should have the component with the loading animation and not UserList.

leekahung replied:
Fixed the ordering, looks good now.

leekahung committed:
Corrected ordering of ternary in UserSection to render loading animation

leekahung reviewed (approved):
Alright, looks good. I believe all the major points from everyone who reviewed it has been touched. Think this is ready for merging.
